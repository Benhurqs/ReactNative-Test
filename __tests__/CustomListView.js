// import 'react-native';
import React from 'react';
import CustomListview from '../app/components/CustomListview';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import Discography from '../app/mock/discography.json';

test('renders correctly', () => {
  const tree = renderer.create(
    <CustomListview 
            itemList={Discography}
        />
  );
  expect(tree).toMatchSnapshot();
});
