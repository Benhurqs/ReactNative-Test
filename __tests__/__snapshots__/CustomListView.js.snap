// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "artist": "Taylor Swift",
          "image": "https://images-na.ssl-images-amazon.com/images/I/61McsadO1OL.jpg",
          "thumbnail_image": "https://i.imgur.com/K3KJ3w4h.jpg",
          "title": "Taylor Swift",
          "url": "https://www.amazon.com/Taylor-Swift/dp/B0014I4KH6",
        },
        Object {
          "artist": "Taylor Swift",
          "image": "https://images-na.ssl-images-amazon.com/images/I/51qmhXWZBxL.jpg",
          "thumbnail_image": "https://i.imgur.com/K3KJ3w4h.jpg",
          "title": "Fearless",
          "url": "https://www.amazon.com/Fearless-Enhanced-Taylor-Swift/dp/B001EYGOEM",
        },
        Object {
          "artist": "Taylor Swift",
          "image": "https://images-na.ssl-images-amazon.com/images/I/51vlGuX7%2BFL.jpg",
          "thumbnail_image": "https://i.imgur.com/K3KJ3w4h.jpg",
          "title": "Speak Now",
          "url": "https://www.amazon.com/Speak-Now-Taylor-Swift/dp/B003WTE886",
        },
        Object {
          "artist": "Taylor Swift",
          "image": "https://images-na.ssl-images-amazon.com/images/I/41j7-7yboXL.jpg",
          "thumbnail_image": "https://i.imgur.com/K3KJ3w4h.jpg",
          "title": "Red",
          "url": "https://www.amazon.com/Red-Taylor-Swift/dp/B008XNZMOU",
        },
        Object {
          "artist": "Taylor Swift",
          "image": "https://images-na.ssl-images-amazon.com/images/I/717DWgRftmL._SX522_.jpg",
          "thumbnail_image": "https://i.imgur.com/K3KJ3w4h.jpg",
          "title": "1989",
          "url": "https://www.amazon.com/1989-Taylor-Swift/dp/B00MRHANNI",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    invertStickyHeaders={undefined}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFF",
              "borderRadius": 5,
              "elevation": 2,
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 8,
              "marginLeft": 16,
              "marginRight": 16,
              "marginTop": 8,
              "padding": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://images-na.ssl-images-amazon.com/images/I/61McsadO1OL.jpg",
              }
            }
            style={
              Object {
                "borderRadius": 50,
                "height": 50,
                "width": 50,
              }
            }
          />
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
                "marginLeft": 12,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000",
                  "fontSize": 16,
                }
              }
            >
              Taylor Swift
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 11,
                  "fontStyle": "italic",
                }
              }
            >
              Taylor Swift
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFF",
              "borderRadius": 5,
              "elevation": 2,
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 8,
              "marginLeft": 16,
              "marginRight": 16,
              "marginTop": 8,
              "padding": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://images-na.ssl-images-amazon.com/images/I/51qmhXWZBxL.jpg",
              }
            }
            style={
              Object {
                "borderRadius": 50,
                "height": 50,
                "width": 50,
              }
            }
          />
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
                "marginLeft": 12,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000",
                  "fontSize": 16,
                }
              }
            >
              Fearless
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 11,
                  "fontStyle": "italic",
                }
              }
            >
              Taylor Swift
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFF",
              "borderRadius": 5,
              "elevation": 2,
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 8,
              "marginLeft": 16,
              "marginRight": 16,
              "marginTop": 8,
              "padding": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://images-na.ssl-images-amazon.com/images/I/51vlGuX7%2BFL.jpg",
              }
            }
            style={
              Object {
                "borderRadius": 50,
                "height": 50,
                "width": 50,
              }
            }
          />
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
                "marginLeft": 12,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000",
                  "fontSize": 16,
                }
              }
            >
              Speak Now
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 11,
                  "fontStyle": "italic",
                }
              }
            >
              Taylor Swift
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFF",
              "borderRadius": 5,
              "elevation": 2,
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 8,
              "marginLeft": 16,
              "marginRight": 16,
              "marginTop": 8,
              "padding": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://images-na.ssl-images-amazon.com/images/I/41j7-7yboXL.jpg",
              }
            }
            style={
              Object {
                "borderRadius": 50,
                "height": 50,
                "width": 50,
              }
            }
          />
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
                "marginLeft": 12,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000",
                  "fontSize": 16,
                }
              }
            >
              Red
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 11,
                  "fontStyle": "italic",
                }
              }
            >
              Taylor Swift
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFF",
              "borderRadius": 5,
              "elevation": 2,
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 8,
              "marginLeft": 16,
              "marginRight": 16,
              "marginTop": 8,
              "padding": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://images-na.ssl-images-amazon.com/images/I/717DWgRftmL._SX522_.jpg",
              }
            }
            style={
              Object {
                "borderRadius": 50,
                "height": 50,
                "width": 50,
              }
            }
          />
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
                "marginLeft": 12,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000",
                  "fontSize": 16,
                }
              }
            >
              1989
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 11,
                  "fontStyle": "italic",
                }
              }
            >
              Taylor Swift
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
